src/App.tsx

```diff
@@ -15,48 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortType: SortType,
+  reverseField = false,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((a, b): number => {
+      switch (sortType) {
+        case SortType.Alphabet:
+          return a.localeCompare(b);
+
+        case SortType.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.None);
+  const [isListReversed, setIsListReversed] = useState(false);
+
+  const filteredGoods = getPreparedGoods(
+    goodsFromServer,
+    sortType,
+    isListReversed,
+  );
+
+  const canWeReset: boolean = (Boolean(sortType) || isListReversed);
+
+  const handlerClick = () => {
+    setIsListReversed(false);
+    setSortType(SortType.None);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortType(SortType.Alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== SortType.Alphabet,
+          })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortType(SortType.Length)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortType !== SortType.Length,
+          })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => setIsListReversed(!isListReversed)}
```

Comment:
```suggestion
          onClick={() => setIsListReversed(isReversed => !isReversed)}
```

If you change state based on previous state you need to use callback for this